Examples & updated to OpenAPI version 3.1.0

donwilson-rezare opened the pull request:
I have updated the files in the repository to OpenAPI version 3.1.0

- replaced "nullable" with null type option
- updated the url-schemes examples to 3.1.0, 
- added the "$schema" tag to resources
- added examples to LandUseActivityType and SpatialResourceType

donwilson-rezare committed:
Update CropType.json

Replaced "nullable", with   "type": ["string", "null"],

donwilson-rezare committed:
Update ResourceType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update AreaMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update MassMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update SpatialMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update MetaDataType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update VelocityMeasureType.json

donwilson-rezare committed:
Update DistanceMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update IrradianceMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update uncefactMassOrVolumeUnits.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update HumidityMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update GeoJSON-example.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update ClassificationType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update LandUseResourceType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update MassPerVolumeMeasurementType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update FeatureCatalogItemResource.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update LengthMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update VolumeMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update TemperatureMeasureType.json

Replaced "nullable": true   with   "type": ["string", "null"]

donwilson-rezare committed:
Update GeoJSON-example.json

Update OpenAPI spec to 3.1.0

donwilson-rezare committed:
Update array-example.json

Update OpenAPI spec to 3.1.0

donwilson-rezare committed:
Update FeatureCatalogItemResource.json

Added $schema dialect tag

donwilson-rezare committed:
Update HoldingResource.json

Added schema dialect tag

donwilson-rezare committed:
Update LandCoverResource.json

Added schema dialect tag

donwilson-rezare committed:
Update OperationRecordResource.json

Added schema dialect tag

donwilson-rezare committed:
Update OperationRecordResource.json

donwilson-rezare committed:
Update FeatureCatalogItemResource.json

donwilson-rezare committed:
Update HoldingResource.json

donwilson-rezare committed:
Update LandCoverResource.json

donwilson-rezare committed:
Update PlaceResource.json

Added schema dialect tag

donwilson-rezare committed:
Update PlotResource.json

Added schema dialect tag

donwilson-rezare committed:
Update PlotZoneResource.json

Added schema dialect tag

donwilson-rezare committed:
Update ProductResource.json

Added schema dialect tag

donwilson-rezare committed:
Update SiteResource.json

Added schema dialect tag

donwilson-rezare committed:
Update WorkRecordResource.json

Added schema dialect tag

donwilson-rezare committed:
Update geojson-agriculturalContentResource.json

Added schema dialect tag

donwilson-rezare committed:
Update SpatialResourceType.json

Example added

donwilson-rezare committed:
Update LandUseActivityType.json

Example added

cookeac commented on enums/uncefactMassOrVolumeUnits.json:
```diff
@@ -3,13 +3,13 @@
        {
           "$ref": "../enums/uncefactMassUnits.json",
-          "nullable": true,
+          "type": ["string", "null"],
```
You cannot specify "type" if using "$ref". The type should be edited inside the referenced enumeration file. You will need to check all other uses of the referenced file to ensure it is ok for the value to be nullable everywhere it is used.

cookeac commented on resources/FeatureCatalogItemResource.json:
```diff
@@ -23,7 +24,7 @@
                     },
                     "description": "Array that is the bounding box coordinates as per GeoJSON",
-                    "nullable": true
+                    "type": ["string", "null"]
```
I don't think this is right. Above "items" is the "type" specifier, which would need to be ["array", "null"] providing this lints ok.

cookeac commented on resources/FeatureCatalogItemResource.json:
```diff
@@ -37,15 +38,15 @@
                     "format": "uri-reference",
                     "description": "The URL from which to fetch a GeoJSON feature collection.",
-                    "nullable": true
+                    "type": ["string", "null"]
```
type is already specified above. That is the line that should be modified.

cookeac reviewed (changes requested):
Changing "type" to an array that can be the specified type or "null" (the null value type) is not a drop-in replacement for "nullable": true. Instead you need to look for where the type is specified and make the appropriate change there.

Commit:
fix: Corrected handling of null values

Changed units enumerations to allow null values (i.e. in uncefactXXXXUnits files).
Removed $schema specification from Resource files.

Removed the erroneous "type": ["string","null"] that I had inserted to replace "nullable" and instead updated existing "type" statements.

cookeac reviewed (approved):
This seems good now. It would be good if anyone is using OpenAPI Codegen if they could check it - esp. in C# (this has been the correct behaviour for most language generators since 2021).
